core/src/main/java/nl/nn/adapterframework/doc/model/FrankElement.java

```diff
@@ -73,13 +75,23 @@ public FrankElement(final String fullName, final String simpleName, boolean isAb
 		this.allChildren = new HashMap<>();
 		this.allChildren.put(FrankAttribute.class, new LinkedHashMap<>());
 		this.allChildren.put(ConfigChild.class, new LinkedHashMap<>());
+		this.xmlElementNames = new ArrayList<>();
 	}
 
 	public void setParent(FrankElement parent) {
 		this.parent = parent;
 		this.statistics = new FrankElementStatistics(this);
 	}
 
+	public void addXmlElementName(String elementName) {
+		// See https://stackoverflow.com/questions/16764007/insert-into-an-already-sorted-list/16764413
+		// for an explanation of this algorithm.
+		int index = Collections.binarySearch(xmlElementNames, elementName, null);
+	    if (index < 0) {
+	        xmlElementNames.add(Misc.binarySearchResultToInsertionPoint(index), elementName);
+	    }
```

Comment:
Hier een call naar InsertIntoSortedArray() gebruiken

Reply:
Ik heb een method gemaakt die zoiets doet en die gebruik ik. Is dat nu dubbelop?